tests/test_cli_actions.py

```diff
@@ -0,0 +1,211 @@
+            status = "Status:              Failed\n"
+            result_message = "Some message2"
+
+            m.setattr(pennylane_orquestra.cli_actions, "workflow_details", lambda *args: status)
+            m.setattr(
+                pennylane_orquestra.cli_actions,
+                "workflow_results",
+                lambda *args: result_message,
+            )
+
+            # Check that looping raises an error if the workflow details
+            # contain a failed status
+            with pytest.raises(
+                ValueError, match=f"Something went wrong with executing the workflow. {status}"
+            ):
+                loop_until_finished("Some ID", timeout=1)
+
+    def test_valid_url(self, monkeypatch, tmpdir):
+        """Test that when receiving a valid url, data will be decoded and
+        returned."""
+        decoded_data = {"res": "Decoded Data"}
+        test_file = os.path.join(tmpdir, "workflow_result.json")
+        test_tar = os.path.join(tmpdir, "test.tgz")
+
+        with open(test_file, "w") as outfile:
+            json.dump(decoded_data, outfile)
+
+        tar = tarfile.open(test_tar, mode="w:gz")
+        tar.add(test_tar)
+        tar.close()
+
+        # Change to the test directory
+        os.chdir(tmpdir)
+        with monkeypatch.context() as m:
+            result_message = ["Some message2", "Some location"]
+
+            m.setattr(
+                pennylane_orquestra.cli_actions, "workflow_results", lambda *args: result_message
+            )
+            m.setattr(urllib.request, "urlopen", lambda arg: arg)
+            m.setattr(urllib.request, "urlretrieve", lambda *args, **kwargs: (test_tar,))
+            assert loop_until_finished("Some ID", timeout=1) == decoded_data
+
+    def test_unexpected_datatype_result(self, monkeypatch, tmpdir):
+        """Test that an error is raised when the result is not a tarfile."""
+        decoded_data = {"res": "Decoded Data"}
+        test_file = os.path.join(tmpdir, "workflow_result.json")
+        test_tar = os.path.join(tmpdir, "test.tgz")
+
+        with open(test_file, "w") as outfile:
+            json.dump(decoded_data, outfile)
+
+        tar = tarfile.open(test_tar, mode="w:gz")
+        tar.add(test_tar)
+        tar.close()
+
+        # Change to the test directory
+        os.chdir(tmpdir)
+        with pytest.raises(ValueError, match="not a tarfile"):
+            with monkeypatch.context() as m:
+                result_message = ["Some message2", "Some location"]
+
+                m.setattr(
+                    pennylane_orquestra.cli_actions, "workflow_results", lambda *args: result_message
+                )
+                m.setattr(urllib.request, "urlopen", lambda arg: arg)
+
+                # test_file is a json file instead of a tarfile
+                m.setattr(urllib.request, "urlretrieve", lambda *args, **kwargs: (test_file,))
+                loop_until_finished("Some ID", timeout=1)
```

Comment:
Where is the `assert` this test?

Reply:
according to the docstring, I expected to find a `pytest.raises` or similar

Reply:
Nevermind, I see it now, it's just very far from the actual function being tested :smile:

Reply:
Could you move the `pytest.raises` to be closer to the last 1-2 lines of the test?

Reply:
For sure! :slightly_smiling_face:  Thanks for the comment, updated the test.